### PR TITLE
[WIP] Group tasks claim

### PR DESCRIPTION
**Pull request by TheHollidayInn**
Fixes put_issue_url_here
### Changes

---

UUID:

### CODE REVIEW

File: website/client-old/js/services/taskServices.js [hunk 1]
```diff
@@ -106,6 +106,21 @@ angular.module('habitrpg')
       });
     };
 
+    function getGroupTasks (groupId) {
```
Comment (paglias) [inline website/client-old/js/services/taskServices.js]: I don't remember, these client side files are from the old PR or are new?

File: website/server/controllers/api-v3/tasks/groups.js [hunk 2]
```diff
@@ -122,7 +122,7 @@ api.assignTask = {
     let group = await Group.getGroup({user, groupId: task.group.id, fields: requiredGroupFields});
     if (!group) throw new NotFound(res.t('groupNotFound'));
 
-    if (group.leader !== user._id) throw new NotAuthorized(res.t('onlyGroupLeaderCanEditTasks'));
+    if (group.leader !== user._id && user._id !== assignedUserId) throw new NotAuthorized(res.t('onlyGroupLeaderCanEditTasks'));
```
Comment (paglias) [inline website/server/controllers/api-v3/tasks/groups.js:125]: this is correct? Does it mean that the user who assign must be the assigned one?

**Comment by paglias** [comment]
For the client side files, if you're adding them to every PR I think it's enough to just remove them from the manifest.json file so they're not loaded in production

File: website/client-old/js/services/taskServices.js (hunk 1 above)
Comment (TheHollidayInn) [reply website/client-old/js/services/taskServices.js]: Eh, these were duplicated during my merge. Let me remove the extra functions.

File: website/server/controllers/api-v3/tasks/groups.js (hunk 2 above)
Comment (TheHollidayInn) [reply website/server/controllers/api-v3/tasks/groups.js:125]: This means that the user has to be the leader or they have to be assigning themselves. 

assignedUserId is the user that is being assigned. So, if the user._id is equal to the user being assigned, they are assigning themselves. 

You can also look at the opposite logic: (!condition1 && !condition2) = (condition1 || condition2) if that helps...

**Comment by TheHollidayInn** [comment]
What is the correct way to add them for every PR? I assumed I would add them to the manifest - let the review happen- then remove them from the manifest.

File: website/server/controllers/api-v3/tasks/groups.js (hunk 2 above)
Comment (paglias) [reply website/server/controllers/api-v3/tasks/groups.js:125]: You're right, sorry I misread that

**Comment by paglias** [comment]
It's fine the way you did it, I got confused by the duplicate functions and thought that you removed all the client side PR and added it back

File: website/views/shared/tasks/edit/advanced_options.jade [hunk 3]
```diff
@@ -13,6 +13,9 @@ div(ng-if='::task.type!="reward"')
     group-tasks-actions(ng-if="obj.type == 'guild'", task='task', group='obj')
   
   div(ng-show='task._edit._advanced')
+    group-tasks-actions
+
+  div(ng-show='task._edit._advanced')
```
Comment (paglias) [inline website/views/shared/tasks/edit/advanced_options.jade:15]: it's already at line 15, it's needed to duplicate it?

Comment (TheHollidayInn) [reply website/views/shared/tasks/edit/advanced_options.jade:15]: Nope! Updated.

**Comment by TheHollidayInn** [comment]
Not sure why the branch failed. Local test run and there is no output.

**Pull request merged**